spec/models/manageiq/providers/embedded_ansible/automation_manager/configuration_script_source_spec.rb

```diff
@@ -1,10 +1,177 @@
+  # using a live repo
+  let(:params) do
+    {
+      :name    => "hello_world",
+      :scm_url => "https://github.com/NickLaMuro/ansible-tower-samples"
+    }
+  end
+
+  let(:repo_dir) { described_class::REPO_DIR }
+  let(:repos)    { Dir.glob(File.join(repo_dir, "*")) }
+
   before do
     EvmSpecHelper.assign_embedded_ansible_role
   end
 
-  it_behaves_like 'ansible configuration_script_source'
+  # Clean up repo dir after each spec
+  after do
+    FileUtils.rm_rf(repos)
+  end
+
+  describe ".create_in_provider" do
+    let(:notify_creation_args) { notification_args('creation') }
+
+    context "with valid params" do
+      it "creates a record and initializes a git repo" do
+        expect(Notification).to receive(:create!).with(notify_creation_args)
+
+        result  = described_class.create_in_provider manager.id, params
+
+        expect(result).to be_an(described_class)
+        expect(result.scm_type).to eq("git")
+        expect(result.scm_branch).to eq("master")
+        expect(File).to exist(File.join(result.send(:repo_dir), ".git"))
+      end
+    end
+
+    context "with invalid params" do
+      it "does not create a record and does not call git" do
+        params[:name]               = nil
+        notify_creation_args[:type] = :tower_op_failure
+
+        expect(AwesomeSpawn).to receive(:run!).never
+        expect(Notification).to receive(:create!).with(notify_creation_args)
+
+        expect do
+          described_class.create_in_provider manager.id, params
+        end.to raise_error(ActiveRecord::RecordInvalid)
+
+        expect(repos).to be_empty
+      end
+    end
+  end
+
+  describe ".create_in_provider_queue" do
+    it "creates a task and queue item" do
+      EvmSpecHelper.local_miq_server
+      task_id = described_class.create_in_provider_queue(manager.id, params)
+      expect(MiqTask.find(task_id)).to have_attributes(:name => "Creating #{described_class::FRIENDLY_NAME} (name=#{params[:name]})")
+      expect(MiqQueue.first).to have_attributes(
+        :args        => [manager.id, params],
+        :class_name  => described_class.name,
+        :method_name => "create_in_provider",
+        :priority    => MiqQueue::HIGH_PRIORITY,
+        :role        => "ems_operations",
+        :zone        => manager.my_zone
+      )
+    end
+  end
+
+  describe "#update_in_provider" do
+    let(:update_params)      { { :scm_branch => "other_branch" } }
+    let(:notify_update_args) { notification_args('update', update_params) }
+
+    context "with valid params" do
+      it "updates the record and initializes a git repo" do
+        record   = build_record
+        repo_dir = record.send(:repo_dir)
+
+        expect(Notification).to receive(:create!).with(notify_update_args)
+
+        result = record.update_in_provider update_params
+
+        expect(result).to be_an(described_class)
+        expect(result.scm_branch).to eq("other_branch")
```

Comment:
Just noticed that I probably should have done a check on the current git branch... maybe something better to be tested when we implement `lib/git_worktree.rb` and tested then.

Note:  A lot of specs in this file were intentionally written to be high level, since we should tests the guts of the `.sync` and helper methods more thoroughly when we do switch to `rugged`.